nondex-common/src/main/java/edu/illinois/nondex/common/SystematicRandom.java

```diff
@@ -0,0 +1,144 @@
+/*
+The MIT License (MIT)
+Copyright (c) 2015 Alex Gyori
+Copyright (c) 2015 Owolabi Legunsen
+Copyright (c) 2015 Darko Marinov
+Copyright (c) 2015 August Shi
+
+
+Permission is hereby granted, free of charge, to any person obtaining
+a copy of this software and associated documentation files (the
+"Software"), to deal in the Software without restriction, including
+without limitation the rights to use, copy, modify, merge, publish,
+distribute, sublicense, and/or sell copies of the Software, and to
+permit persons to whom the Software is furnished to do so, subject to
+the following conditions:
+
+The above copyright notice and this permission notice shall be
+included in all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+
+package edu.illinois.nondex.common;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Random;
+import java.util.Stack;
+import java.util.logging.Level;
+
+public class SystematicRandom extends Random {
+    private final Stack<ExplorationEntry> choices;
+    private final String logFileName;
+    private final Configuration config = Configuration.parseArgs();
+    private List<String> lines;
+    private int replayIndex;
+
+    public SystematicRandom() {
+        logFileName = config.systematicLog;
+        choices = new Stack<ExplorationEntry>();
+        File file = new File(logFileName);
+        if (file.exists()) {
+            try {
+                lines = Files.readAllLines(Paths.get(logFileName));
+            } catch (IOException ioe) {
+                Logger.getGlobal().log(Level.SEVERE,"Could not read lines from systematic.log" ,ioe);
+            }
+            Object[] choiceValues;
+            for (String element: lines) {
+                String delimiter = "[ ]+";
+                if (!element.isEmpty()) {
+                    choiceValues = element.split(delimiter);
+                    if (choiceValues.length == 3) {
+                        int current = Integer.parseInt(choiceValues[0].toString());
```

Comment:
Since your array will be String[] you will not need to call toString(). Similar on maximum, and shouldExplore.